Sync/State.swift

```diff
@@ -405,13 +401,7 @@ open class Scratchpad {
         self.clientGUID = Bytes.generateGUID()
         self.clientName = DeviceInfo.defaultClientName()
 
-        self.fxaDeviceId = {
-            guard let string = prefs.stringForKey(PrefDeviceRegistration) else {
-                return nil
-            }
-            let json = JSON(parseJSON: string)
```

Comment:
Urgh, gross.

Reply:
Moved to near where `fxaDeviceId` is dug out of prefs, below.